File: backend/app/schemas/message.py
from datetime import datetime
from pydantic import BaseModel


class AttachmentOut(BaseModel):
    id: str
    file_name: str
    mime_type: str
    file_size_bytes: int

    model_config = {"from_attributes": True}


class ReplySnippet(BaseModel):
    message_id: str
    sender_name: str
    content_preview: str  # first 200 chars of original message


class MessageOut(BaseModel):
    id: str
    conversation_id: str
    sender_id: str
    sender_name: str
    content: str
    message_type: str  # "text" | "image" | "file"
    reply_to: ReplySnippet | None
    attachments: list[AttachmentOut]
    created_at: datetime

File: frontend/types/message.ts
export interface AttachmentOut {
  id: string;
  file_name: string;
  mime_type: string;
  file_size_bytes: number;
}

export interface ReplySnippet {
  message_id: string;
  sender_name: string;
  content_preview: string;
}

export interface Message {
  id: string;
  conversation_id: string;
  sender_id: string;
  sender_name: string;
  content: string;
  message_type: "text" | "image" | "file";
  reply_to: ReplySnippet | null;
  attachments: AttachmentOut[];
  created_at: string; // ISO datetime
  receipt_status: "sending" | "sent" | "delivered" | "read";
}
